uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -577,13 +577,12 @@ define([
     function initTime (dateType) {
       var time = moment(vm.request[dateType + '_date']).format('HH:mm');
       var timeObject = vm.uiOptions.times[dateType];
-      var isBeforeWorkPatternRange = getTimeDifferenceInHours(timeObject.min, time) <= 0;
-      var isAfterWorkPatternRange = getTimeDifferenceInHours(timeObject.max, time) >= 0;
+      var isOutsideWorkPatternRange =
+        getTimeDifferenceInHours(timeObject.min, time) <= 0 ||
+        getTimeDifferenceInHours(timeObject.max, time) >= 0;
 
-      if (dateType === 'from' && isBeforeWorkPatternRange) {
-        time = timeObject.min;
-      } else if (dateType === 'to' && isAfterWorkPatternRange) {
-        time = timeObject.max;
+      if (isOutsideWorkPatternRange) {
+        time = timeObject[dateType === 'from' ? 'min' : 'max'];
```

Comment:
You keep using that pattern which is interesting, but it's harder to read (even if you save a few characters). This is easier to read:

```js
time = dateType === 'from' ? timeObject.min : timeObject.max
```

Because instead of having `object -> condition -> propertyA or propertyB`  you have ` condition -> object.propertyA or object.propertyB` which follows more naturally the relationship between object and properties.

Reply:
Done.